ask-sdk-core/ask_sdk_core/utils/request_util.py

```diff
@@ -0,0 +1,301 @@
+    :rtype: str
+    """
+    return handler_input.request_envelope.context.system.device.device_id
+
+
+def get_dialog_state(handler_input):
+    # type: (HandlerInput) -> Optional[DialogState]
+    """Return the dialog state enum from the intent request.
+
+    The method retrieves the `dialogState` from the intent request, if
+    the skill's interaction model includes a dialog model. This can be
+    used to determine the current status of user conversation and return
+    the appropriate dialog directives if the conversation is not yet complete.
+    More information on dialog management can be found here :
+    https://developer.amazon.com/docs/custom-skills/define-the-dialog-to-collect-and-confirm-required-information.html
+
+    The method returns a ``None`` if there is no dialog model added or
+    if the intent doesn't have dialog management. The method raises a
+    :py:class:`TypeError` if the input is not an `IntentRequest`.
+
+    :param handler_input: The handler input instance that is generally
+        passed in the sdk's request and exception components.
+    :type handler_input: ask_sdk_core.handler_input.HandlerInput
+    :return: State of the dialog model from the intent request.
+    :rtype: Optional[ask_sdk_model.dialog_state.DialogState]
+    :raises: TypeError if the input is not an IntentRequest
+    """
+    request = handler_input.request_envelope.request
+    if isinstance(request, IntentRequest):
+        return request.dialog_state
+
+    raise TypeError("The provided request is not an IntentRequest")
+
+
+def get_slot(handler_input, slot_name):
+    # type: (HandlerInput, AnyStr) -> Optional[Slot]
+    """Return the slot information from intent request.
+
+    The method retrieves the slot information
+    :py:class:`ask_sdk_model.slot.Slot` from the input intent request
+    for the given ``slot_name``. More information on the slots can be
+    found here :
+    https://developer.amazon.com/docs/custom-skills/request-types-reference.html#slot-object
+
+    If there is no such slot, then a ``None``
+    is returned. If the input request is not an
+    :py:class:`ask_sdk_model.intent_request.IntentRequest`, a
+    :py:class:`TypeError` is raised.
+
+    :param handler_input: The handler input instance that is generally
+        passed in the sdk's request and exception components
+    :type handler_input: ask_sdk_core.handler_input.HandlerInput
+    :param slot_name: Name of the slot that needs to be retrieved
+    :type slot_name: str
+    :return: Slot information for the provided slot name if it exists,
+        or a `None` value
+    :rtype: Optional[ask_sdk_model.slot.Slot]
+    :raises: TypeError if the input is not an IntentRequest
+    """
+    request = handler_input.request_envelope.request
+    if isinstance(request, IntentRequest):
+        if request.intent.slots is not None:
+            return request.intent.slots.get(slot_name, None)
+        else:
```

Comment:
Is the else condition required here? Since you are already passing None as default value.